api/base/serializers.py

```diff
@@ -1837,3 +1839,27 @@ class MaintenanceStateSerializer(ser.ModelSerializer):
     class Meta:
         model = MaintenanceState
         fields = ('level', 'message', 'start', 'end')
+
+
+class HideIfSwitch(ConditionalField):
+    """
+    If switched is switched this field is hidden.
```

Comment:
Should state what direction the switch should be: On/Off or True/False

Reply:
I intentional left this ambigous because turning on a flag could mean turning a field off and vice versa. It's set with the `hide_if` parameter [here](https://github.com/CenterForOpenScience/osf.io/pull/9275/files/260bed375f36e0a372c08ad0f5931efffdd92135#diff-4b396d38a0947821679352d308445c5aR1848)

Reply:
Cool. Maybe update the docstring to make that more clear. "Hide this field if the switch state matches the `hide_if` parameter" or similar?